Fix incorrect dropdown menu location for the archive button

GetToSet opened the pull request:
<!--
Issues and PRs are typically reviewed Wednesday and most weekend mornings.
-->

This PR fixes the incorrect dropdown menu location for the archive button.
<img width="1213" alt="Screenshot 2022-08-12 23 51 03" src="https://user-images.githubusercontent.com/8158163/184394178-3020574b-7c3a-47aa-8c84-a7c3608540b5.png">

Both dropdown menus for the flag and dropdown buttons should work well after this fix:
<img width="922" alt="Screenshot 2022-08-12 23 52 37" src="https://user-images.githubusercontent.com/8158163/184394608-abc72b57-fc8f-4a88-a087-6cafc6d2d156.png">

<img width="922" alt="Screenshot 2022-08-12 23 52 34" src="https://user-images.githubusercontent.com/8158163/184394618-873070a5-2d55-48a0-a60b-39af9f632010.png">

ymeynot45 commented:
First, thank you for finding that bug as I introduced it when I refactored the flag dropdown to remove the jQuery. Additionally, it is awesome that you found a solution so fast.

I am just going to give two notes that will get your pull request approved faster.

1) The site is trying to move away from inline styling.  If you want to get ahead of the upcoming change requests, move it to a css file and add classes to the flagging span & the archive span.
2) I would also convert it back to rems.  Building sites for people with vision problems has become a priority and rems [make it easier for people with disabilities to use the site](https://github.com/lobsters/lobsters/issues/1123.). 
The flag dropdown needs -2rem and the archive needs -0.125rem to align them to the previous lobsters px standard.

Once again, thank you for fixing my mistakes.

GetToSet commented:
> First, thank you for finding that bug as I introduced it when I refactored the flag dropdown to remove the jQuery. Additionally, it is awesome that you found a solution so fast.
> 
> I am just going to give two notes that will get your pull request approved faster.
> 
> 1. The site is trying to move away from inline styling.  If you want to get ahead of the upcoming change requests, move it to a css file and add classes to the flagging span & the archive span.
> 2. I would also convert it back to rems.  Building sites for people with vision problems has become a priority and rems [make it easier for people with disabilities to use the site](https://github.com/lobsters/lobsters/issues/1123).
>    The flag dropdown needs -2rem and the archive needs -0.125rem to align them to the previous lobsters px standard.
> 
> Once again, thank you for fixing my mistakes.

Hi @ymeynot45, I've removed inline styling by adding a new CSS class `dropdown_parent`. This should be a common class for all drop-downs. I've also updated px to rems, and could you provide some more context about why rems are better than px in the context of accessibility?

ymeynot45 commented:
Hey thanks for replying:
So when people with vision problems change the size of the font, so they
can see it, the whole layout goes out the window if you use px. Px is a
static number that doesn't change based on the user settings.  But rems are
based off of the font size the user has set as their default.  So if the
user has the font set to 125% then rems will see that and change everything
to be unscaled to 125%.  This helps keep things inline based on the user
settings.

This is a nice article that covers it.
https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility/

It will show that in this particular case it isn't that important.  But it
is good to get people in the habit of using rems except where they
shouldn't.  Just like you use "const" instead of "let" except when you
shouldn't. I'll admit that it is a little pedantic.  But that is how
progress is made.

Thank you, once again for replying, your work is greatly appreciated.
- YmeYnot

On Fri, Aug 19, 2022 at 2:24 AM Ethan Wong ***@***.***> wrote:

> First, thank you for finding that bug as I introduced it when I refactored
> the flag dropdown to remove the jQuery. Additionally, it is awesome that
> you found a solution so fast.
>
> I am just going to give two notes that will get your pull request approved
> faster.
>
>    1. The site is trying to move away from inline styling. If you want to
>    get ahead of the upcoming change requests, move it to a css file and add
>    classes to the flagging span & the archive span.
>    2. I would also convert it back to rems. Building sites for people
>    with vision problems has become a priority and rems make it easier for
>    people with disabilities to use the site
>    <https://github.com/lobsters/lobsters/issues/1123>.
>    The flag dropdown needs -2rem and the archive needs -0.125rem to align
>    them to the previous lobsters px standard.
>
> Once again, thank you for fixing my mistakes.
>
> Hi @ymeynot45 <https://github.com/ymeynot45>, I've removed inline styling
> by adding a new CSS class dropdown_parent. This should be a common class
> for all drop-downs. I've also updated px to rems, and could you provide
> some more context about why rems are better than px in the context of
> accessibility?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lobsters/lobsters/pull/1121#issuecomment-1220341202>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUOCFGFFF4LDDINTMOYKSLVZ4ZDFANCNFSM56MDWUJA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

pushcx commented:
@GetToSet Thanks for finding this bug and fixing it, getting a PR is basically the best kind of bug report there is. :D

@ymeynot45 Thanks for guiding this PR with your familiarity from our js/css work, I agree with your notes.

I'm really glad to merge this fix, cheers!

ymeynot45 commented:
LOL, I remember when I originally fixed that the first time.  I wonder what
changed that broke it again.
Thank you for catching the new bug.
Just a note, the reason I started using rem, it is better for people with
vision problems.

On Fri, Aug 12, 2022 at 10:53 AM Ethan Wong ***@***.***>
wrote:

> This PR fixes the incorrect dropdown menu location for the archive button.
> [image: Screenshot 2022-08-12 23 51 03]
> <https://user-images.githubusercontent.com/8158163/184394178-3020574b-7c3a-47aa-8c84-a7c3608540b5.png>
>
> Both dropdown menus for the flag and dropdown buttons should work well
> after this fix:
> [image: Screenshot 2022-08-12 23 52 37]
> <https://user-images.githubusercontent.com/8158163/184394608-abc72b57-fc8f-4a88-a087-6cafc6d2d156.png>
>
> [image: Screenshot 2022-08-12 23 52 34]
> <https://user-images.githubusercontent.com/8158163/184394618-873070a5-2d55-48a0-a60b-39af9f632010.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/lobsters/lobsters/pull/1121
> Commit Summary
>
>    - bdc1b32
>    <https://github.com/lobsters/lobsters/pull/1121/commits/bdc1b32f2ac836ef5b5a32b9553c1ab40af4efc7>
>    Fix incorrect dropdown menu location for the archive button
>
> File Changes
>
> (3 files <https://github.com/lobsters/lobsters/pull/1121/files>)
>
>    - *M* app/assets/javascripts/application.js.erb
>    <https://github.com/lobsters/lobsters/pull/1121/files#diff-23509090366c6e92e805951c16d5c7c36d364b58bbe72730c8dd524b9d38a041>
>    (9)
>    - *M* app/assets/stylesheets/application.css.erb
>    <https://github.com/lobsters/lobsters/pull/1121/files#diff-373a35818d62d883962d4ee806dd20b35d25a95ea4ef24e324721cc72c508787>
>    (2)
>    - *M* app/views/stories/_listdetail.html.erb
>    <https://github.com/lobsters/lobsters/pull/1121/files#diff-de456492c7e130e5c0e8b08ae60e12931affba64d025e9ac1fbe9f7fa388c6cc>
>    (7)
>
> Patch Links:
>
>    - https://github.com/lobsters/lobsters/pull/1121.patch
>    - https://github.com/lobsters/lobsters/pull/1121.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lobsters/lobsters/pull/1121>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUOCFARMFHW3V2AOLRLN53VYZXQVANCNFSM56MDWUJA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
